### PR TITLE
Adding test to cover default serialization behavior

### LINKED ISSUE #350: Add tests for correct null handling
Need to add test coverage to ensure regression like [this ](https://github.com/Azure/azure-cosmos-dotnet-v3/issues/346) doesn't happen again.

### PR DESCRIPTION
**Pull request by ealsur**
# Default serialization test

## Description

This PR adds a test to assert the default serialization behavior (`NullValueHandling = NullValueHandling.Ignore`) and make sure we don't regress it in any future change.

## Type of change

- [X] New feature (non-breaking change which adds functionality)

## Closing issues

This PR closes #350

### CODE REVIEW

**Comment by simplynaveen20** [comment]
While working on the LINQ tests , existing V2 test were failing which were expecting null object , I am working on item where i changed this behavior and removed NullValueHandling.Ignore from the default serializer . Shouldn't the default of serialization  allow null if user is setting it ?

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosJsonSerializerTests.cs [hunk 1]
```diff
@@ -64,6 +64,30 @@ public async Task TestCustomJsonSerializer()
...
+                cost = double.MaxValue
+            };
+
+            await container.UpsertItemAsync(document);
+
+            CosmosResponseMessage cosmosResponseMessage = await container.ReadItemAsStreamAsync(new PartitionKey(document.status), document.id);
+            StreamReader reader = new StreamReader(cosmosResponseMessage.Content);
+            string text = reader.ReadToEnd();
+
+            Assert.AreEqual(-1, text.IndexOf("description"), "Stored document contains a null attribute");
```
Comment (simplynaveen20) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosJsonSerializerTests.cs]: "description" -> nameof(document.description)

**Comment by ealsur** [comment]
> While working on the LINQ tests , existing V2 test were failing which were expecting null object , I am working on item where i changed this behavior and removed NullValueHandling.Ignore from the default serializer . Shouldn't the default of serialization allow null if user is setting it ?

If we change that behavior, we just adapt the test. The goal of the test is not to do any accidental regressions, but a conscious one.

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosJsonSerializerTests.cs (hunk 1 above)
Comment (ealsur) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosJsonSerializerTests.cs]: Done

**Comment by simplynaveen20** [comment]
> > While working on the LINQ tests , existing V2 test were failing which were expecting null object , I am working on item where i changed this behavior and removed NullValueHandling.Ignore from the default serializer . Shouldn't the default of serialization allow null if user is setting it ?
> 
> If we change that behavior, we just adapt the test. The goal of the test is not to do any accidental regressions, but a conscious one.

Make sense , please go forward with this , and if i change the behavior in my work item  i will juts flip the assert

**Comment by kirankumarkolli** [comment]
Thanks @ealsur
